jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/CheckboxTests.java

```diff
@@ -1,57 +1,97 @@
+import static io.github.com.pages.CheckBoxPage.indeterminateCheckbox;
+import static io.github.com.pages.CheckBoxPage.primaryColorCheckbox;
+import static io.github.com.pages.CheckBoxPage.requiredCheckbox;
+import static io.github.com.pages.CheckBoxPage.resultCheckbox;
+import static io.github.com.pages.CheckBoxPage.warnColorCheckbox;
 
-// TODO Move to the new page
-@Ignore
 public class CheckboxTests extends TestsInit {
     @BeforeMethod(alwaysRun = true)
     public void before() {
-        shouldBeLoggedIn();
-        angularPage.open();
-        disabledCheckbox.show();
+        checkBoxPage.open();
+        waitCondition((()  -> checkBoxPage.isOpened()));
+        checkBoxPage.checkOpened();
     }
 
-    @Test
+    @Test(description = "Test verifies label value of checkbox")
     public void checkLabelValue() {
         basicCheckbox.label().has().value("Check me!");
         resultCheckbox.label().has().value("I'm a checkbox");
     }
 
-    @Test
+    @Test(description = "Test verifies initial checkbox state")
     public void basicCheckboxValidation() {
+        basicCheckbox.show();
         basicCheckbox.is().displayed().and().enabled().and().deselected();
     }
 
-    @Test
+    @Test(description = "Test verifies checkbox selection state")
     public void checkBasicCheckbox() {
         basicCheckbox.check();
         basicCheckbox.is().selected();
     }
 
-    @Test
+    @Test(description = "Test verifies checkbox feature: indeterminate")
     public void indeterminateTest() {
-        indeterminateCheckbox.click();
+        indeterminateCheckbox.show();
+        indeterminateCheckbox.check();
         resultCheckbox.is().indeterminate();
     }
 
-    @Test
+    @Test(description = "Test verifies checkbox disabled state")
     public void disabledOption() {
-        disabledCheckbox.click();
+        disabledCheckbox.check();
         resultCheckbox.is().disabled();
     }
 
-    @Test
+    @Test(description = "Test verifies checkbox align positions")
     public void configurableCheckboxTest() {
-        indeterminateCheckbox.check();
         alignBeforeRadioButton.click();
-        resultCheckbox.is().indeterminate().and().cssClass("mat-checkbox-label-before");
+        resultCheckbox.is().alignedBefore();
+        alignAfterRadioButton.click();
+        resultCheckbox.is().alignedAfter();
+    }
+
+    @Test(description = "Test verifies that checkbox has no ripple effect")
+    public void noRippleEffect() {
+        disabledCheckbox.show();
+        disableRippleCheckbox.is().noRippleEffect();
     }
+
+    @Test(description = "Test verifies checkbox aria-label")
+    public void ariaLabel() {
+        ariaLabelCheckbox.show();
+        String ARIALABEL = "First checkbox";
+        ariaLabelCheckbox.is().hasAriaLabel(ARIALABEL);
+    }
+
+    @Test(description = "Test verifies checkbox feature: required option")
+    public void isRequired() {
+        requiredCheckbox.show();
+        resultCheckbox.is().isRequired();
```

Comment:
Why/how resultCheckbox - became"reqired"?

Please integrate a "Checkbox should be checked!" verification. 

Also is this message is configured in an angular element itself? Should we add some predefined method to check it like .getWarningMessage() - just for checkBox with required attribute? Just to think about it.

Reply:
Besides testing presense of attribute 'required' I added testing of 'text field' with notification. And that after 'required checkbox' is checked that text field does not exist.